src/util/init_ini_data.cpp

```diff
@@ -451,8 +452,14 @@ namespace hpx { namespace util
             return plugin_registries;
 
         // make sure each node loads libraries in a different order
+#if defined(HPX_HAVE_CXX11_STD_SHUFFLE)
+        std::random_device random_device;
+        std::mt19937 generator(random_device());
+        std::shuffle(libdata.begin(), libdata.end(), std::move(generator));
+#else
```

Comment:
In my experience, `std::random_device` is a bigger hurdle to compatibility... It's one of the reasons we normally use the boost counterpart in other places. But I have not checked recently whether the situation has changed.